edge-modules/DMSender/src/Program.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace DMSender
+{
+    using System;
+    using System.Globalization;
+    using System.IO;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+
+    class Program
+    {
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Console.WriteLine($"[{DateTime.UtcNow.ToString("MM/dd/yyyy hh:mm:ss.fff tt", CultureInfo.InvariantCulture)}] Main()");
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("config/appsettings.json", optional: true)
+                .AddEnvironmentVariables()
+                .Build();
+
+            TimeSpan dmDelay = configuration.GetValue("DMDelay", TimeSpan.FromSeconds(5));
+
+            string targetModuleId = configuration.GetValue("TargetModuleId", "DMReceiver");
+
+            // Get deviced id of this device, exposed as a system variable by the iot edge runtime
+            string targetDeviceId = System.Environment.GetEnvironmentVariable("IOTEDGE_DEVICEID");
```

Comment:
configuration.GetValue can be used because the env variables are added to configuration